python/ray/tests/test_generators.py

```diff
@@ -179,6 +180,32 @@ def static(num_returns):
         ray.get(static.remote(3))
 
 
+def test_dynamic_generator_distributed(ray_start_cluster):
+    cluster = ray_start_cluster
+    # Head node with no resources.
+    cluster.add_node(num_cpus=0)
+    ray.init(address=cluster.address)
+    cluster.add_node(num_cpus=1)
+    cluster.wait_for_nodes()
+
+    @ray.remote(num_returns="dynamic")
+    def dynamic_generator(num_returns):
+        for i in range(num_returns):
+            yield np.ones(1_000_000, dtype=np.int8) * i
+            time.sleep(0.1)
```

Comment:
how often does this test reproduce the bug? If low, would increase the sleep increase the likelyhood of reproducibility?

Reply:
It reproduces reliably on my laptop! So I think it's probably safe enough on CI.